fix(UIShell): reduce padding on header name to fix action button (#11687)

asamaiya00 opened the pull request:
UIShell: SideNav Rail w/header - reduce padding on header name to fix the width of menu action button

by: Animesh Samaiya (asamaiya00@gmail.com)

Closes #11687 

Menu button was becoming squashed somewhere between sm breakpoint (320px) and md breakpoint (672px). Fixed it by removing extra padding when md breakpoint is active. Also set the minimum width of the menu button, so that it doesn't squash further.

#### Changelog

**Changed**

packages/styles/scss/components/ui-shell/header/_header.scss

#### Testing / Reviewing

Current: 
![image](https://user-images.githubusercontent.com/58809223/194153381-79461a5f-b4d3-4761-b2e9-b5a3869711f8.png)

Changed:
![image](https://user-images.githubusercontent.com/58809223/194153526-b3bcfc64-556b-4668-9b73-2e0a8e1514f8.png)

Please review and let me know of any changes. 

@tw15egan

asamaiya00 committed:
fix(UIShell): reduce padding on header name to fix action button(#11687)

UIShell: SideNav Rail w/header - reduce padding to fix width of action button

by: Animesh Samaiya (asamaiya00@gmail.com)

tw15egan commented on packages/styles/scss/components/ui-shell/header/_header.scss:
```diff
@@ -42,6 +42,9 @@
     transition: background-color $duration-fast-02,
       border-color $duration-fast-02;
+    @include breakpoint-down('md') {
+      min-width: mini-units(6);
```
Could we use the spacing tokens here? 

https://carbondesignsystem.com/guidelines/spacing/overview/#spacing-scale

asamaiya00 replied:
Let me try that

asamaiya00 replied:
Yup, done @tw15egan

tw15egan commented on packages/styles/scss/components/ui-shell/header/_header.scss:
```diff
@@ -137,6 +140,9 @@
     transition: border-color $duration-fast-02;
     user-select: none;
+    @include breakpoint-down('md') {
+      padding: 0 mini-units(2);
```
Could we use the spacing tokens here? 

https://carbondesignsystem.com/guidelines/spacing/overview/#spacing-scale

tw15egan reviewed (commented):
Just a couple comments

asamaiya00 committed:
Merge branch 'main' into bug/UI-Shell-menu-button-became-squashed-on-sm-breakpoint-#11687

asamaiya00 committed:
fix(UIShell): use spacing tokens (#11687)

UIShell: SideNav Rail w/header - reduce padding to fix width of action button

    by: Animesh Samaiya (asamaiya00@gmail.com)

asamaiya00 committed:
Merge branch 'bug/UI-Shell-menu-button-became-squashed-on-sm-breakpoint-#11687' of https://github.com/asamaiya00/carbon into bug/UI-Shell-menu-button-became-squashed-on-sm-breakpoint-#11687

aledavila committed:
Merge branch 'main' into bug/UI-Shell-menu-button-became-squashed-on-sm-breakpoint-#11687

kodiakhq[bot] committed:
Merge branch 'main' into bug/UI-Shell-menu-button-became-squashed-on-sm-breakpoint-#11687
